File: 05- May/05- Maximum Number of Vowels in a Substring of Given Length/05- Maximum Number of Vowels in a Substring of Given Length (Noura Algohary).py
# Author: Noura Algohary
# Use sliding window 

class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowelsCnt = 0
        
        for i in range(k):
            if s[i] in ['a', 'e', 'i', 'o', 'u']:
                vowelsCnt += 1
        
        maxVowels = vowelsCnt

        for i in range(k, len(s)):
            if s[i] in ['a', 'e', 'i', 'o', 'u']:
                vowelsCnt += 1
            
            if s[i - k] in ['a', 'e', 'i', 'o', 'u']:
                vowelsCnt -= 1
            
            if vowelsCnt > maxVowels:
                maxVowels = vowelsCnt
        
        return maxVowels

File: 05- May/05- Maximum Number of Vowels in a Substring of Given Length/05- Maximum Number of Vowels in a Substring of Given Length (Noura Algohary).cpp
// Author: Noura Algohary
//  Use sliding window 
class Solution {
public:
    int maxVowels(string s, int k) {
        int n = s.size(), vowelsCnt = 0;

        for(int i = 0; i < k ; i++)
        {
            if(s[i] == 'a' || s[i] == 'e' ||s[i] == 'i' ||s[i] == 'o' ||s[i] == 'u')
                vowelsCnt++;

        }
        int maxVowels = vowelsCnt;
        for(int i = k; i < n; i++)
        {
            if(s[i] == 'a' || s[i] == 'e' ||s[i] == 'i' ||s[i] == 'o' ||s[i] == 'u')
                ++vowelsCnt;

            if(s[i - k] == 'a' || s[i - k] == 'e' ||s[i - k] == 'i' ||s[i - k] == 'o' ||s[i - k] == 'u')
                --vowelsCnt;

            if(vowelsCnt > maxVowels)
                maxVowels = vowelsCnt;
        }
        return maxVowels;
    }
};